### PR TITLE
Introduce supervisor object

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Adds a new `supervisor` object that holds information that is passed down to the panels.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: hassio/src/supervisor-base-element.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+declare global {
+  interface HASSDomEvents {
+    "supervisor-update": Partial<Supervisor>;
+  }
+}
+
+export class SupervisorBaseElement extends urlSyncMixin(
+  ProvideHassLitMixin(LitElement)
+) {
+  @property({ attribute: false }) public supervisor!: Supervisor;
```
Comment (bramkragten) [inline hassio/src/supervisor-base-element.ts:26]: It is not always defined, so we should either give it a default value or replace the `!` with an `?`

File: hassio/src/dashboard/hassio-update.ts [hunk 2]
```diff
@@ -74,26 +64,26 @@ export class HassioUpdate extends LitElement {
         <div class="card-group">
           ${this._renderUpdateCard(
             "Home Assistant Core",
-            this.hassInfo!,
+            this.supervisor.core!,
```
Comment (bramkragten) [inline hassio/src/dashboard/hassio-update.ts:67]: This could be undefined

Comment (ludeeus) [reply hassio/src/dashboard/hassio-update.ts:67]: How?

File: src/data/supervisor/supervisor.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+} from "../hassio/supervisor";
+
+export interface Supervisor {
+  host: HassioHostInfo;
+  supervisor: HassioSupervisorInfo;
+  info: HassioInfo;
+  core: HassioHomeAssistantInfo;
+  network: NetworkInfo;
+  resolution: HassioResolution;
+  os: HassioHassOSInfo;
```
Comment (bramkragten) [inline src/data/supervisor/supervisor.ts:17]:
```suggestion
  os?: HassioHassOSInfo;
```

File: src/data/supervisor/supervisor.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  HassioInfo,
+  HassioSupervisorInfo,
+} from "../hassio/supervisor";
+
+export interface Supervisor {
+  host: HassioHostInfo;
+  supervisor: HassioSupervisorInfo;
+  info: HassioInfo;
+  core: HassioHomeAssistantInfo;
+  network: NetworkInfo;
```
Comment (bramkragten) [inline src/data/supervisor/supervisor.ts:15]:
```suggestion
  network?: NetworkInfo;
```

File: src/data/supervisor/supervisor.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  resolution: HassioResolution;
```
Comment (bramkragten) [inline src/data/supervisor/supervisor.ts:16]:
```suggestion
  resolution?: HassioResolution;
```

File: hassio/src/dashboard/hassio-update.ts [hunk 6]
```diff
@@ -74,26 +64,26 @@ export class HassioUpdate extends LitElement {
...
             "hassio/homeassistant/update",
             `https://${
-              this.hassInfo?.version_latest.includes("b") ? "rc" : "www"
+              this.supervisor.core?.version_latest.includes("b") ? "rc" : "www"
```
Comment (bramkragten) [inline hassio/src/dashboard/hassio-update.ts:70]:
```suggestion
              this.supervisor.core.version_latest.includes("b") ? "rc" : "www"
```

File: hassio/src/dashboard/hassio-update.ts [hunk 7]
```diff
@@ -74,26 +64,26 @@ export class HassioUpdate extends LitElement {
...
             }.home-assistant.io/latest-release-notes/`
           )}
           ${this._renderUpdateCard(
             "Supervisor",
-            this.supervisorInfo!,
+            this.supervisor.supervisor!,
```
Comment (bramkragten) [inline hassio/src/dashboard/hassio-update.ts:75]:
```suggestion
            this.supervisor.supervisor,
```

File: hassio/src/dashboard/hassio-update.ts [hunk 8]
```diff
@@ -74,26 +64,26 @@ export class HassioUpdate extends LitElement {
...
             "hassio/supervisor/update",
             `https://github.com//home-assistant/hassio/releases/tag/${
-              this.supervisorInfo!.version_latest
+              this.supervisor.supervisor!.version_latest
```
Comment (bramkragten) [inline hassio/src/dashboard/hassio-update.ts:78]:
```suggestion
              this.supervisor.supervisor.version_latest
```

File: hassio/src/dashboard/hassio-update.ts (hunk 2 above)
Comment (bramkragten) [reply hassio/src/dashboard/hassio-update.ts:67]: You are right, so we can drop the `!` :-)

File: hassio/src/supervisor-base-element.ts (hunk 1 above)
Comment (ludeeus) [reply hassio/src/supervisor-base-element.ts:26]:
```suggestion
  @property({ attribute: false }) public supervisor?: Supervisor;
```

**Review by balloob — APPROVED** [review]

**Pull request merged**